src/kernel/mm.h

```diff
@@ -0,0 +1,22 @@
+
+#ifndef _MM_H_
+# define _MM_H_
+
+#include "types.h"
+
+#define MEMORY_START (void*) 0x1000 // Beggining of dynamically allocated memory by the kernel
+#define MEMORY_END (void*) 0xE999 // End of dynamically allocated memory by the kernel (we don't want to interfere with the kernel stack, do we?)
+#define TOTAL_MEMORY MEMORY_END - MEMORY_START
+#define MEMORY_OWNER_FREE 0xFFFF
+#define MEMORY_MIN_SIZE 16
+
+struct KmallocHeader {
+    u16 size;
+    u16 user;
+};
+
+void initMm();
+void *kmalloc(u16 owner, u16 size);
+void kfree(void *addr);
```

Comment:
I think these functions should be the  only thing in the .h, the rest should be in the .c